datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -1025,7 +1025,53 @@ pub fn to_substrait_rex(
             col_ref_offset,
             extension_info,
         ),
-        _ => not_impl_err!("Unsupported expression: {expr:?}"),
+        Expr::IsNull(arg) => {
+            let arguments: Vec<FunctionArgument> = vec![FunctionArgument {
+                arg_type: Some(ArgType::Value(to_substrait_rex(
+                    arg,
+                    schema,
+                    col_ref_offset,
+                    extension_info,
+                )?)),
+            }];
+
+            let function_name = "is_null".to_string();
+            let function_anchor = _register_function(function_name, extension_info);
```

Comment:
Very minor, but why not call this `function_reference` to match the field name used below?

Reply:
While reviewing the code again, I found this simply follows the same pattern as the existing substrait code, so looks good to me

Reply:
@tgujar  if you hav time, it would also be awesome if you could make a PR that renames this variable (and other uses of `_register_function` to `function_reference` which I think would make the code cleaner